src/main/java/org/jboss/ejb/protocol/remote/RemotingEJBDiscoveryProvider.java

```diff
@@ -403,13 +418,36 @@ void connectAndDiscover(URI uri, String clusterEffective) {
                 return;
             }
             outstandingCount.getAndIncrement();
+
+            // keep a record of this URI if it has an associated cluster (EJBCLIENT-325)
+            if (clusterEffective != null) {
+                urisByCluster.computeIfAbsent(clusterEffective, ignored -> Collections.newSetFromMap(new ConcurrentHashMap<>())).add(uri);
+            }
```

Comment:
It seems to me that we are only creating the entry for the first node. What about a scenario that we are connecting to another node of a cluster but there is already an entry for that custer in urisByCluster map. Shouldn't we add the node there as well?

Reply:
This actually does allow multiple entries to be added. If i'm not mistaken, if we call computeIfAbsent and there is no Set<URI> value associated with the key, it will be created and returned; otherwise, the existing value will be returned. In either case, the new uri will be added to the Set returned as a value. I'm also seeing sets with multiple URI entries in my logs.

Reply:
yeah, you are right